### PR TITLE
AccessControl: Enable RBAC by default

### PR DESCRIPTION
**Pull request by IevaVasiljeva**
**What this PR does / why we need it**:
Enables RBAC by default:
* add a new configuration section for RBAC in Grafana config;
* remove RBAC feature toggle;
* fix some tests that require a bit more extra setup with RBAC.

**Which issue(s) this PR fixes**:

Related to https://github.com/grafana/grafana-enterprise/issues/2503

**Special notes for your reviewer**:
related enterprise PR: https://github.com/grafana/grafana-enterprise/pull/3274

### CODE REVIEW

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 1]
```diff
@@ -2447,8 +2447,8 @@ func TestEval(t *testing.T) {
 				}
 			}
 			`,
-			expectedStatusCode: http.StatusBadRequest,
-			expectedMessage:    "invalid condition: invalid query A: data source not found: unknown",
+			expectedStatusCode: http.StatusUnauthorized,
+			expectedMessage:    "user is not authorized to query one or many data sources used by the rule",
```
Comment (IevaVasiljeva) [inline pkg/tests/api/alerting/api_alertmanager_test.go:2451]: Note: here (and in a few other cases) the returned error message differs, we should check with the alerting team if it's ok.

File: pkg/tests/api/alerting/api_alertmanager_test.go [hunk 2]
```diff
@@ -2650,25 +2650,20 @@ func TestEval(t *testing.T) {
 
 // createFolder creates a folder for storing our alerts under. Grafana uses folders as a replacement for alert namespaces to match its permission model.
 // We use the dashboard command using IsFolder = true to tell it's a folder, it takes the dashboard as the name of the folder.
-func createFolder(t *testing.T, store *sqlstore.SQLStore, folderID int64, folderName string) (string, error) {
+func createFolder(t *testing.T, folderUID, grafanaListedAddr, login, password string) error {
```
Comment (IevaVasiljeva) [inline pkg/tests/api/alerting/api_alertmanager_test.go:2653]: Note: calling the dashboard store directly does not set up the right permissions, so I've changed it to call the endpoint and use the full folder creation flow.

File: pkg/tests/testinfra/testinfra.go [hunk 3]
```diff
@@ -316,4 +322,5 @@ type GrafanaOpts struct {
 	DisableLegacyAlerting                 bool
 	EnableUnifiedAlerting                 bool
 	UnifiedAlertingDisabledOrgs           []int64
+	DisableRBAC                           bool
```
Comment (IevaVasiljeva) [inline pkg/tests/testinfra/testinfra.go:325]: Note: required for alerting tests with RBAC disabled, but if we manage to make all the tests run fine with RBAC (which would be great), we can remove this.

File: pkg/tests/api/alerting/api_prometheus_test.go [hunk 4]
```diff
@@ -717,6 +720,7 @@ func TestPrometheusRulesPermissions(t *testing.T) {
 	}
 
 	// remove permissions from folder2
+	// TODO this too
```
Comment (IevaVasiljeva) [inline pkg/tests/api/alerting/api_prometheus_test.go:723]: Note: I forgot to address this - we should change these folder ACL updates to the corresponding RBAC updates.

**Review by Jguer — APPROVED** [review]

**Pull request merged**